test/unit-tests/vast_test.js

```diff
@@ -1721,4 +1761,281 @@ describe('ad_manager_vast', function() {
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    expect(companion).to.eql(parsedAd.companion);
+  });
+
+  it('VPAID 2.0: Ad should not end on adVideoEnded', function() {
+    var podEndNotified = 0, linearEndNotified = 0;
+    vpaidInitialize();
+
+    amc.notifyPodEnded = function() {
+      podEndNotified++;
+    };
+
+    amc.notifyLinearAdEnded = function() {
+      linearEndNotified++;
+    };
+
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    vastAdManager.adVideoEnded();
+    expect(podEndNotified).to.eql(0);
+    expect(linearEndNotified).to.eql(0);
+  });
+
+  it('VPAID 2.0: Ad Unit should handle clickthru if playerHandles is false, otherwise players handle the click', function() {
+    var adUnitHandling = true;
+    vpaidInitialize();
+
+    vastAdManager.openUrl = function(url) {
+      adUnitHandling = false;
+    };
+
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    ad.vpaidAd.sendClick(false);
+    expect(adUnitHandling).to.be(true);
+
+    ad.vpaidAd.sendClick(true);
+    expect(adUnitHandling).to.be(false);
+  });
+
+  it('VPAID 2.0: Should notify linear ad started when adLinearChange is sent', function() {
+    var linearStartedNotified = 0;
+    vpaidInitialize(vpaidNonLinearXML);
+
+    amc.notifyLinearAdStarted = function() {
+      linearStartedNotified++;
+    };
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    amc.adManagerSettings['linearAdSkipButtonStartTime'] = 5;
+    ad.vpaidAd.sendAdLinearChange(false);
+    expect(linearStartedNotified).to.eql(0);
+    ad.vpaidAd.sendAdLinearChange(true);
+   expect(linearStartedNotified).to.eql(1);
+  });
+
+  it('VPAID 2.0: Should parse and send ad parameters', function() {
+    vpaidInitialize();
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    expect(JSON.parse(ad.ad.adParams)).to.eql(ad.vpaidAd.properties.adParameters);
+  });
+
+  it('VPAID 2.0: Should hide player ui', function() {
+    var hidePlayerUi = false;
+    amc.hidePlayerUi = function() {
+      hidePlayerUi = true;
+    };
+    vpaidInitialize();
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    expect(hidePlayerUi).to.be(true);
+  });
+
+  it('VPAID 2.0: Should resize ad unit on size changed', function() {
+    var hidePlayerUi = false;
+    vpaidInitialize();
+    var ad = amc.timeline[0];
+    vastAdManager.playAd(ad);
+    vastAdManager.initializeAd();
+    expect(ad.vpaidAd.properties.width).to.be(100);
+    expect(ad.vpaidAd.properties.height).to.be(100);
+    vastAdManager._slot.offsetWidth = 200;
+    vastAdManager._slot.offsetHeight = 200;
+    amc.publishPlayerEvent(amc.EVENTS.SIZE_CHANGED);
+    expect(ad.vpaidAd.properties.width).to.be(200);
+    expect(ad.vpaidAd.properties.height).to.be(200);
```

Comment:
Can we use different values for width and height? This will allow us to detect if there is a width/height mixup in our logic in the future.